src/button-refactor/ButtonsLinksAPI.md

```diff
@@ -0,0 +1,68 @@
+- More design discussion, particularly focused on outline button
+[x] https://github.com/github/primer/issues/272
+- Make icon only buttons square
+[x] https://github.com/github/primer/issues/321
+- Super specific button use case?
+[x] https://github.com/github/primer/issues/350
+- a11y audit with task list for PVC
+[x] https://github.com/github/primer/issues/382
+- React button refactor
+[x] https://github.com/github/primer/issues/468
+- Visual bugs (colors, state)
+[x] https://github.com/github/primer/discussions/87
+- Discussion about accessory buttons
+[x] https://github.com/github/primer/discussions/211
+- Open ended discussion about primary button as dropdown
+[x] https://github.com/github/primer/discussions/459
+- Allie's thoughts on limiting icon only buttons and working towards encouraging visual labels
+[x] https://github.com/github/primer/discussions/477
+- Buttons styled as links, links styled as buttons
+
+## Specific notes from previous bugs to keep in mind
+[] Check all button variants have shadow present
+[] Ensure icon colors are consistent in hover states
+[] Ensure icon colors are consistent with variants in all states
+[] Ensure disabled colors are consistent across frameworks
+
+## Component list
+A few discussions were about naming and prop drilling (source here). We found that for Button to handle all of its use cases, we would need a large number of props– some conditional and dependent on one another. When that happens, it's typically a sign that the logic should be separated into another component.
+
+| Component | Description |
+| -- | -- |
+| Button | standard `button` with variants, size, visual slots |
+| IconButton | `button` with icon only (square) and required `aria-label` |
+| ButtonStyledAsLink | `button` that visually looks like a link |
+| ReactionButton | `button` snowflake with specific styles/interaction design |
+| ButtonGroup | wrapper to handle grouping buttons |
+| Link | `a` with variants, optional trailing visuals |
+| LinkStyledAsButton | `a` with button variants, required trailing visuals |
+
+## Component API breakdown
+
+### Button
+
+| prop | type | options | default | notes |
+| -- | -- | -- | -- | -- |
+| `variant` | one-of string | `primary` `secondary` `danger` `outline`? | `secondary` | |
+| `size` | one-of string | `small` `default` `large` | `default | |
+| `label` | string | button description | null | |
+| `aria-label` | string | button description for screen readers | null | |
+| `aria-pressed` | boolean | `true/false` | `false` | |
+| `leadingVisual` | children (slot) | octicon | null | |
+| `trailingVisual` | children (slot) | octicon | null | |
+| `trailingAction` | children (slot) | octicon | null | slot for caret to maintain leading/trailing visuals if they exist |
+| `fullWidth` | boolean | `true/false` | `false` | |
+| `visualPosition` | one-of string | `fixed` `some-other-word` | `fixed` | lock icon to the text label or to the button container |
```

Comment:
Do we need an option for alignment with text? I'd imagine that you'd sometimes want `ButtonStyledAsLink` to be inline and other times want it to line up with other buttons 🥴

Reply:
Ahh see that's where I also get 🥴 

I feel like `ButtonStyledAsLink` means you want to hide the fact that this is actually a button.. so its inline, blue text, not centered text.. but retains height for accessibility

Which makes me think we still need a subtle Button variant. When you want a Button that's.. subtle 🙃

Reply:
I guess there are two use cases for buttons styled as links:
1. Secondary CTAs in forms or modals, e.g.
  ![Modal with dismiss link-style button](https://user-images.githubusercontent.com/1901935/149920855-6851f0bc-2298-4610-a6cc-125e035cf46b.png)
1. Downplaying visual weight, where they should always be accompanied by an icon/action, and are in line with text e.g.
  ![Assignees gear icon with dropdown action](https://user-images.githubusercontent.com/1901935/149921614-71bb6a80-20da-4bbc-86d4-93cfb68871cb.png)
  ![Add assignees button with dropdown action](https://user-images.githubusercontent.com/1901935/149921772-39ccc4f7-9388-4a4c-ae7d-d6214f1d7539.png)

And some ideas for how to accommodate both of those cases:
1. The first one is some subtle/invisible `Button` variant and the second is a `ButtonStyledAsLink` (so BSAL always requires a trailing icon/action)
  It feels a bit weird that you could achieve a similar visual style for button elements using different combinations of components/props, I feel like that's how we got into this mess?
1. They're both `ButtonStyledAsLink` but we have a display prop to toggle between block/inline
  Then it's a bit ambiguous when icon/action are required, only when they're not locked up with other buttons?
1. They're both `ButtonStyledAsLink` and are inline, but we use a button group component for vertical alignment of locked-up buttons

FWIW I think they can be in line with text but have padding and negative margin so they still have accessible tap targets, but maybe that's a different conversation.

Also curious how `<summary>` comes in to this discussion, if at all 🥴 

Far too many thoughts for one comment, I'm sorry 🙃